odoo/addons/base/models/ir_profile.py

```diff
@@ -108,3 +118,25 @@ def set_profiling(self, profile=None, collectors=None, params=None):
             'collectors': request.session.profile_collectors,
             'params': request.session.profile_params,
         }
+
+
+class EnableProfilingWizard(models.TransientModel):
+    _name = 'base.enable.profiling.wizard'
+    _description = "Enable profiling for some time"
+
+    duration = fields.Selection([
+        ('minutes_5', "5 Minutes"),
+        ('hours_1', "1 Hour"),
+        ('days_1', "1 Day"),
+        ('months_1', "1 Month")
+    ], string="Enable profiling for")
+    expiration = fields.Datetime("Enable profiling until", compute='_compute_expiration', store=True, readonly=False)
+
+    @api.depends('duration')
+    def _compute_expiration(self):
+        for record in self:
+            unit, quantity = (record.duration or 'days_0').split('_')
+            record.expiration = fields.Datetime.now() + relativedelta(**{unit: int(quantity)})
```

Comment:
Nice trick, but it makes the code fragile.

Reply:
I stopped reading after the comma :D (Except if you have annoter suggestion)

Reply:
Guess the mapping could be done by hand, probably resulting in a no-op if the stored value is incorrect rather than e.g. an error.

On the other hand, it would allow updating the durations more easily, here the "old durations" will remain stored in the database which is probably not great if e.g. we decide to update 5mn to 10 because 5 is a bit short.

Maybe the keys could be either just the unit or an abstract value which gets mapped at execution time?

Then again it's just a transient wizard so it's not like the records should live for ages.